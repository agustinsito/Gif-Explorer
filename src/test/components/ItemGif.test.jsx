const { render, screen } = require("@testing-library/react")
const { ItemGif } = require("../../components/ItemGif")

describe('pruebas en ItemGif', () => { 

    const title = 'Qiyana'

    const url = 'http://qiyana-sito.com/qiyana.jpg'

    test('Debe matchear el snapshot.', () => { 
        const { container } = render(<ItemGif title= { title } url ={ url }/>)
        expect ( container ).toMatchSnapshot()
     })

    test('Mostrar la imagen con el alt y url indicado. ', () => { 
        render(<ItemGif title= { title } url ={ url }/>)
        // screen.debug(); //  renders on cls the current sintaxis 

        // expect( screen.getByRole('img').src).toBe( url ) 
        // expect( screen.getByRole('img').alt).toBe( title ) 
        
        const { src, alt } = screen.getByRole('img')
        expect   ( src ). toBe ( url )
        expect   ( alt ). toBe ( title )

     })

 })