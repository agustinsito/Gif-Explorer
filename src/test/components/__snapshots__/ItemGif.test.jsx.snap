// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en ItemGif Debe matchear el snapshot. 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Qiyana"
      src="http://qiyana-sito.com/qiyana.jpg"
    />
    <p>
      Qiyana
    </p>
  </div>
</div>
`;
